### DISCUSSION TITLE
Cannot enter anything :-(

### DISCUSSION THREAD

**Discussion by BrendanMK** [DamianMorozov/OpenTgResearcher#23] [Q&A]
Hi Damian,
I used to work with this one before: https://github.com/Dineshkarthik/telegram_media_downloader
And there I found TgDownloader.

Even though I'm an IT-guy, I'm obviously not clever enough to get it running. What I get is this:
  

                                                       Main menu
╭─────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────╮
│ Name                    │ Value                                                                                      │
├─────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────┤
│ ✓ App version           │ v.0.1.610                                                                                  │
│ x Application settings  │ Something is need setup                                                                    │
│ ✓ Storage settings      │ Everything is ok                                                                           │
│ x Client settings       │ Something is need setup                                                                    │
│ x Download settings     │ Something is need setup                                                                    │
╰─────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────╯
  Switch menu number

> Exit
  Application settings
  Storage settings
  Client settings
  Download settings
  Advanced

My system:
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22621 N/A Build 22621
.NET                            7.0.3  x64

EDIT:
I can walk through the items with arrow down/up.
But if, e.g. I try to 'create new storage', there is nothing I can enter.
And when I select 'delete existing storage' and press ENTER, CMD exits.

Some fields you cannot leave until you enter something. I finally got it running, tried to save some data, but I have to enter part of my 'credentials' every time I start TgDownloader again.

I finally managed to start a download 'manually', but the first 80 % of messages weren't downloaded at all, then message 257 til 273 was downloaded, but only videos, no images.

And the download stopped in the middle of video 274, while the total number was 299.

**NOTE:** [Dineshkarthik/telegram_media_downloader](https://github.com/Dineshkarthik/telegram_media_downloader) downloads everything, also images and it downloads all messages from 1 til 299. However, telegram_media_downloader needs several runs until it catches all files: It remembers the ones, which could not be downloaded and only repeats download attempts of those files.

So, here are some open questions:
- How to I save my 'credentials' so that I don't have to enter my phone number or my api_hash again every time I start TgDownloader?
- How do I download images or other files?
- ...

BTW
The xml file reads this:

```
<App>
  <FileSession>TgDownloader.session</FileSession>
  <FileStorage>TgDownloader.db</FileStorage>
</App>
```

Thanks and best regards,
Brendan

**Comment by DamianMorozov** [comment]
You look like a good quality tester. It's nice to see such an extended message.

I also, like you, have used the above project. It is really good, except that some of the files occasionally fail to download, or have 0 size.

Regarding the re-request of your authorization data. It looks like you have some problems with the storage and session creation. Check if these files (`TgDownloader.session`, `TgDownloader.db`) exist in your file system, in the same directory where you unzipped `TgDownloader`.

**Comment by DamianMorozov** [comment]
```
But if, e.g. I try to 'create new storage', there is nothing I can enter.
And when I select 'delete existing storage' and press ENTER, CMD exits.
```
You are right, when you try to delete the repository, there is an error and a termination. But, when you start the program, the storage file `TgDownloader.db` is created automatically. If it doesn't, more information is needed. I will consider adding some error logs that you could send to me for analysis.

**Reply by BrendanMK** [reply]
Thanks for the reply.
What about those two issues:
It did not start with the first message, but with message 257 and stopped (crashed?) in the middle of the download of message 273.
Total number of messages is 299.

And other questions I had:

Let me answer to this first:
`Regarding the re-request of your authorization data. It looks like you have some problems with the storage and session creation. Check if these files (TgDownloader.session, TgDownloader.db) exist in your file system, in the same directory where you unzipped TgDownloader.`

Yes, they do exist there!

- So, my 'credentials' (phone number and api_hash) should be saved? Can I enter them manually to some file?
- Only videos where downloaded, no images. And some text from the messages I found in TgDownloader.db. How do I download images or other files? Can I define file extensions? Or can I force downloading other files. At the moment, I just tested with one single channel
- ...
- 
Please feel free to assign some tasks to me, what I should test.

PS: Maybe you could send me example files, how they should look like, "dummy" credentials included.
File that were created in my test runs are:
TgDownloader.db
TgDownloader.session
TgDownloader.xml
